### ISSUE TITLE
SA1312 is raised for deconstructed tuple elements when SA1316 is set to PascalCase

### ISSUE THREAD

**Issue by 0x084E** [DotNetAnalyzers/StyleCopAnalyzers#3031]
The following code raises SA1312 for the tuple element names

```
var properties = new List<(string Name, string Value)>();
foreach ((string Name, string Value) in properties) // SA1312 Raised
{
}
```

But if i change the names to camelCase SA1316 is raised:
```
var properties = new List<(string Name, string Value)>();
foreach ((string name, string value) in properties) // SA1316 Raised
{
}
```

I'm not sure which one is right 🤔

**Comment by sharwell** [comment]
Camel case would be correct here, even if tuples are set to Pascal case.

**Comment by 0x084E** [comment]
OK, that also most obvious to me und the rule I already follow 😄

**Comment by sharwell** [comment]
@vweijsters I already wrote a bunch of tests for this
